_rules/SC4-1-2-role-attribute-has-valid-value.md

```diff
@@ -0,0 +1,114 @@
+---
+name: Role attribute has valid value
+test_type: atomic
+
+description: |
+   This rule checks that, elements with a role attribute have valid values
+   
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+Any [non-empty](#non-empty) `role` attribute that is placed on an HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies).
+
+### Expectation
+
+Each test target has at least one valid value as specified in the [WAI-ARIA 1.1, roles](https://www.w3.org/TR/wai-aria-1.1/#roles_categorization).
```

Comment:
Should be "non-abstract valid role". This should also allow roles from the DPUB and Graphics modules. I think maybe we should create a definition for that. Something like "current ARIA specifications". WDYT?

Reply:
Good point.

Reply:
Should there be something here about the separate values being space-separated?

Reply:
@annethyme - I do not believe, we need to detail that when two words are separated by space, that they are separate values.

Reply:
Created an issue to create definition - `non-abstract valid role` - https://github.com/auto-wcag/auto-wcag/issues/277, so this PR can progress.